basyx.submodelservice/basyx.submodelservice-core/src/main/java/org/eclipse/digitaltwin/basyx/submodelservice/value/PropertyValue.java

```diff
@@ -47,4 +47,12 @@ public PropertyValue(String value) {
 	public String getValue() {
 		return value;
 	}
+	
+	@Override
+    public boolean equals(Object obj) {
```

Comment:
do we need to override equals here?

Reply:
In getSubmodelByIdValueOnlyExistingSubmodel, I need to compare the objects of type SubmodelValueOnly which have a Map of items where the value is of type SubmodelElementValue. 

PropertyValue implements the interface SubmodelElementValue, hence, when i compare the SubmodelValueOnly objects, I'm actually comparing their map of items to check if they are equal. 

Given that PropertyValue isn't a primitive data type, I had to add the equals method so that the comparison can be made between the items. The test fails otherwise.